geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/AvroSimpleFeatureWriter.scala

```diff
@@ -72,12 +69,24 @@ class AvroSimpleFeatureWriter(sft: SimpleFeatureType)
 
     // Write out fields from Simple Feature
     var i = 2
-    while(i < schema.getFields.length) {
+    while(i < schema.getFields.size()) {
       val f = schema.getFields.get(i)
       write(f.schema, f)
       i += 1
     }
   }
 
+  def writeWithUserData(datum: SimpleFeature, out: Encoder) = {
+    defaultWrite(datum, out)
+
+    // TODO move out and use everywhere
+    val aw = new AvroWriter(out)
+    aw.writeGenericMap(datum.getUserData)
+  }
+
+  private val doWrite = if (opts.withUserData) writeWithUserData _ else defaultWrite _
```

Comment:
Thanks.  Is this even better?

val write: (SimpleFeature, Encoder) => Unit = if (opts.withUserData) writeWithUserData else defaultWrite